Library/Homebrew/extend/os/mac/caveats.rb

```diff
@@ -6,44 +6,53 @@ class Caveats
 
   def plist_caveats
     s = []
-    if f.plist || keg&.plist_installed?
-      plist_domain = f.plist_path.basename(".plist")
+    if !f.plist && !f.service? && !keg&.plist_installed?
+      caveat = "#{s.join("\n")}\n" if s.present?
+      return caveat
+    end
```

Comment:
```suggestion
    return if !f.plist && !f.service? && !keg&.plist_installed?
```

I'm pretty sure I'm understanding this right in that if those conditions are met we just want to return `nil` (which is technically done at the moment because `s.present?` will always be `false` so `caveat` will be `nil`.

Reply:
If those are met and there is no previous caveat.

Reply:
It does'n't actually check that here, though. `s` is defined two lines above.

The output of the `plist_caveats` method is added to the rest of the caveats here in `caveats.rb`:

https://github.com/Homebrew/brew/blob/0e27ec707b7c20c7aaa6b27918b597bccd1b93c9/Library/Homebrew/caveats.rb#L43-L45

Note the `.compact` which removes any `nil` values such as an empty `plist_caveats` return value.